Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp

```diff
@@ -2897,7 +2897,14 @@ class ObjectAddingAndRemovingItself : public ThreadSafeRefCountedAndCanMakeThrea
 
     ~ObjectAddingAndRemovingItself()
     {
-        EXPECT_TRUE(m_set.contains(*this));
+        static size_t i { 0 };
+        i++;
+        if (i == 8) {
```

Comment:
I think we would have dropped this `EXPECT_TRUE(m_set.contains(*this));` check instead of giving special treatment to `i ==8`

Reply:
I think we should make ThreadSafeWeakHashSet::contains to return false if the weak pointer is null, but that is unrelated to the amortization algorithm, so I'm planning to do it in a separate PR.

Reply:
Following up in https://github.com/WebKit/WebKit/pull/16512